File: 20170508-typescript-file-logger/app/file-logger.ts
import FileSystem from "./file-system";

export default class FileLogger {

    constructor(private fileSystem: FileSystem) {
    }

    log(text: string) {
        const fileName = this.buildFileName();

        if (this.fileSystem.exists(fileName)) {
            this.fileSystem.append(fileName, text);
            return;
        }

        this.fileSystem.write(fileName, text);
    }

    private buildFileName() {
        const date = new Date();
        const year = date.getFullYear();
        let month = '' + (date.getMonth() + 1);
        let day = '' + date.getDate();

        if (month.length < 2) { month = '0' + month; }
        if (day.length < 2) { day = '0' + day; }

        return `log${year}${month}${day}.txt`;
    }

}

File: 20170520-java-file-logger/src/main/java/com/katas/FileLogger.java
package com.katas;

import com.sun.javafx.binding.StringFormatter;

import java.io.IOException;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class FileLogger {

    private static final String FILE_NAME_TEMPLATE = "log%s.txt";
    private static final String DATE_FORMAT = "yyyyMMdd";

    private FileSystemAdapter fileSystem;
    private DateProvider dateProvider;

    public FileLogger(FileSystemAdapter fileSystemAdapter, DateProvider dateProvider) {
        this.fileSystem = fileSystemAdapter;
        this.dateProvider = dateProvider;
    }

    public void log(String message) throws IOException {
        String fileName = buildFileName();

        if (fileSystem.exists(fileName)) {
            fileSystem.append(fileName, message);
            return;
        }

        fileSystem.write(fileName, message);
    }

    private String buildFileName() {
        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DATE_FORMAT);
        String date = dateProvider.now().format(formatter);

        return String.format(FILE_NAME_TEMPLATE, date);
    }

}
